src/server/raft_client.rs

```diff
@@ -288,14 +289,19 @@ impl Buffer for MessageBuffer {
 }
 
 /// Reporter reports whether a snapshot is sent successfully.
-struct SnapshotReporter<T> {
+struct SnapshotReporter<T, E> {
     raft_router: T,
     region_id: u64,
     to_peer_id: u64,
     to_store_id: u64,
+    _engine: PhantomData<E>,
 }
 
-impl<T: RaftStoreRouter<RocksEngine> + 'static> SnapshotReporter<T> {
+impl<T, E> SnapshotReporter<T, E>
+where
+    T: RaftStoreRouter<E> + 'static,
```

Comment:
Could it be `RaftStoreRouter<impl KvEngine >`?

Reply:
I looked into it and impl trait is not valid in this position.